addOns/alertFilters/src/main/java/org/zaproxy/zap/extension/alertFilters/AlertFilter.java

```diff
@@ -172,13 +278,74 @@ protected static AlertFilter decode(int contextId, String encodedString) {
             alertFilter.setRuleId(Integer.parseInt(pieces[1]));
             alertFilter.setNewRisk(Integer.parseInt(pieces[2]));
             alertFilter.setUrl(new String(Base64.decodeBase64(pieces[3])));
-            alertFilter.setRegex(Boolean.parseBoolean(pieces[4]));
+            alertFilter.setUrlRegex(Boolean.parseBoolean(pieces[4]));
             alertFilter.setParameter(new String(Base64.decodeBase64(pieces[5])));
+            if (pieces.length > 6) {
+                // Older versions will not have included these fields
+                alertFilter.setParameterRegex(Boolean.parseBoolean(pieces[6]));
+                alertFilter.setAttack(new String(Base64.decodeBase64(pieces[7])));
+                alertFilter.setAttackRegex(Boolean.parseBoolean(pieces[8]));
+                alertFilter.setEvidence(new String(Base64.decodeBase64(pieces[9])));
+                alertFilter.setEvidenceRegex(Boolean.parseBoolean(pieces[10]));
+            }
         } catch (Exception ex) {
             log.error("An error occured while decoding alertFilter from: " + encodedString, ex);
             return null;
         }
         // log.debug("Decoded alertFilter: " + alertFilter);
         return alertFilter;
     }
+
+    public boolean appliesToAlert(Alert alert) {
+        if (!isEnabled()) {
+            log.debug("Filter disabled");
+            return false;
+        }
+        if (getRuleId() != alert.getPluginId()) {
+            // rule ids dont match
+            log.debug(
```

Comment:
This and the others in `matchesStringOrRegex` could check if debug is enabled (to avoid creating the strings).